tests/integration/TaxCategory/TaxCategoryUpdateRequestTest.php

```diff
@@ -3,213 +3,193 @@
-
-        $key = $this->getTestRun() . '-new-key';
-        $request = TaxCategoryUpdateRequest::ofIdAndVersion($taxCategory->getId(), $taxCategory->getVersion())
-            ->addAction(
-                TaxCategorySetKeyAction::ofKey($key)
-            )
-        ;
-        $response = $request->executeWithClient($this->getClient());
-        $result = $request->mapResponse($response);
-
-        $this->deleteRequest->setVersion($result->getVersion());
-
-        $this->assertInstanceOf(TaxCategory::class, $result);
-        $this->assertNotSame($key, $draft->getKey());
-        $this->assertSame($key, $result->getKey());
-        $this->assertNotSame($taxCategory->getVersion(), $result->getVersion());
+        $client = $this->getApiClient();
+
+        TaxCategoryFixture::withUpdateableDraftTaxCategory(
+            $client,
+            function (TaxCategoryDraft $draft) {
+                return $draft->setKey('set-test-key');
+            },
+            function (TaxCategory $taxCategory) use ($client) {
+                $key = 'new-test-key';
+
+                $request = RequestBuilder::of()->taxCategories()->update($taxCategory)
+                    ->addAction(TaxCategorySetKeyAction::ofKey($key));
+                $response = $this->execute($client, $request);
+                $result = $request->mapFromResponse($response);
+
+                $this->assertInstanceOf(TaxCategory::class, $result);
+                $this->assertNotSame($key, 'set-test-key');
+                $this->assertSame($key, $result->getKey());
+                $this->assertNotSame($taxCategory->getVersion(), $result->getVersion());
+
+                return $result;
+            }
+        );
     }
 
     public function testSetDescription()
     {
-        $draft = $this->getDraft('set-description');
-        $taxCategory = $this->createTaxCategory($draft);
-
-
-        $description = $this->getTestRun() . '-new-description';
-        $request = TaxCategoryUpdateRequest::ofIdAndVersion(
-            $taxCategory->getId(),
-            $taxCategory->getVersion()
-        )
-            ->addAction(TaxCategorySetDescriptionAction::of()->setDescription($description))
-        ;
-        $response = $request->executeWithClient($this->getClient());
-        $result = $request->mapResponse($response);
-        $this->deleteRequest->setVersion($result->getVersion());
-
-        $this->assertInstanceOf(TaxCategory::class, $result);
-        $this->assertSame($description, $result->getDescription());
-        $this->assertNotSame($taxCategory->getVersion(), $result->getVersion());
+        $client = $this->getApiClient();
+
+        TaxCategoryFixture::withUpdateableDraftTaxCategory(
+            $client,
+            function (TaxCategoryDraft $draft) {
+                return $draft->setDescription('set-description');
+            },
+            function (TaxCategory $taxCategory) use ($client) {
+                $description = 'new-description';
```

Comment:
and if we replace the test variable with something like ?
$description = Uuid::uuidv4() . 'new-description';

Reply:
Maybe better with `'new-description-' . Uuid::uuidv4()`

Reply:
as you proposed in the other PRs, I'm replacing with 
 'new-description-' . TaxCategoryFixture::uniqueTaxCategoryString();